### ISSUE TITLE
Barostat issues with frozen atoms

### ISSUE THREAD

**Issue by pclabaut** [cp2k/cp2k#1591]
Hello,

Trying some MD runs, I have found that the coordinate scaling induced by NPT ensemble usage does affect the position of frozen atoms, and therefore, inter-atomic distances that should be fixed as well. It does not seem to be the desired behaviour in my opinion. 

I have worked on a fix for my own runs but it seems to provoke numerous failures in the test cases. My question is the following: should the npt_i and npt_f be properly fixed (which would probably imply more changes than what I did, as well as modifying some test cases), should I just implement my current version as a new keyword, or should I simply keep my version for myself ?

In any case, I will be happy to provide my working materials but my development time is limited since my PhD is about to come to an end.

Let me know which path you prefer !

Cheers,

Paul Clabaut

**Comment by juerghutter** [comment]
Hi

I think both versions are of interest. It is important that
the input description clearly describes the behavior (absolute
vs relative fixed coordinates).
There are some possible pitfalls for absolute fixed coordinates
with variable cell (e.g. two atoms further apart than the box size),
but that would be the responsability of the user.

regards

Juerg Hutter
--------------------------------------------------------------
Juerg Hutter                         Phone : ++41 44 635 4491
Institut für Chemie C                FAX   : ++41 44 635 6838
Universität Zürich                   E-mail: ***@***.***
Winterthurerstrasse 190
CH-8057 Zürich, Switzerland
---------------------------------------------------------------

-----"pclabaut" ***@***.***> wrote: -----
To: "cp2k/cp2k" ***@***.***>
From: "pclabaut" ***@***.***>
Date: 07/15/2021 09:24AM
Cc: "Subscribed" ***@***.***>
Subject: [cp2k/cp2k] Barostat issues with frozen atoms (#1591)


 Hello,
 Trying some MD runs, I have found that the coordinate scaling induced by NPT ensemble usage does affect the position of frozen atoms, and therefore, inter-atomic distances that should be fixed as well. It does not seem to be the desired behaviour in my opinion.
 I have worked on a fix for my own runs but it seems to provoke numerous failures in the test cases. My question is the following: should the npt_i and npt_f be properly fixed (which would probably imply more changes than what I did, as well as modifying some test cases), should I just implement my current version as a new keyword, or should I simply keep my version for myself ?
 In any case, I will be happy to provide my working materials but my development time is limited since my PhD is about to come to an end.
 Let me know which path you prefer !
 Cheers,
 Paul Clabaut
